ios/RNSScreen.mm

```diff
@@ -391,72 +599,183 @@ @implementation RNSScreen {
   int _dismissCount;
   BOOL _isSwiping;
   BOOL _shouldNotify;
+#endif
 }
 
+#pragma mark - Common
+
 - (instancetype)initWithView:(UIView *)view
 {
   if (self = [super init]) {
     self.view = view;
+#ifdef RN_FABRIC_ENABLED
+    if ([view isKindOfClass:[RNSScreenView class]]) {
```

Comment:
Is there a case when this `if` fails?

Reply:
I'm not aware of such case in our code base, but `initWithView:` instance method is exposed publicly in `RNSScreen` interface so it is not wrong idea to make sure that the right type is passed at runtime. However my guess is that this check was added in `RNSScreenController` to make sure that no one passes there a `RNSScreenView` (it required `RNSScreenComponentView` before the merge happened`. We can either extend this `if statement` for both architecture or remove it completely. What do you think?

Reply:
If we did not check it previously, I think there is no need to do it now as well so we can remove it.